sdk/core/azure_core_amqp/src/error.rs

```diff
@@ -28,20 +39,81 @@ pub enum AmqpErrorKind {
+    //    SenderError(AmqpSenderError),
     TransportImplementationError(Box<dyn std::error::Error + Send + Sync>),
 }
 
-#[derive(Debug, Clone)]
+create_extensible_enum!(
+    #[doc = "AMQP protocol defined error conditions"]
+    AmqpErrorCondition,
+    (DecodeError, "amqp:decode-error"),
+    (FrameSizeTooSmall, "amqp:frame-size-too-small"),
+    (IllegalState, "amqp:illegal-state"),
+    (InternalError, "amqp:internal-error"),
+    (InvalidField, "amqp:invalid-field"),
+    (NotAllowed, "amqp:not-allowed"),
+    (NotFound, "amqp:not-found"),
+    (NotImplemented, "amqp:not-implemented"),
+    (PreconditionFailed, "amqp:precondition-failed"),
+    (ResourceDeleted, "amqp:resource-deleted"),
+    (ResourceLimitExceeded, "amqp:resource-limit-exceeded"),
+    (ResourceLocked, "amqp:resource-locked"),
+    (UnauthorizedAccess, "amqp:unauthorized-access"),
+    (LinkStolen, "amqp:link:stolen"),
+    (LinkPayloadSizeExceeded, "amqp:link:message-size-exceeded"),
+    (LinkDetachForced, "amqp:link:detach-forced"),
+    (ConnectionForced, "amqp:connection:forced"),
+    (ServerBusyError, "com.microsoft:server-busy"),
+    (ArgumentError, "com.microsoft:argument-error"),
+    (
+        ArgumentOutOfRangeError,
+        "com.microsoft:argument-out-of-range"
+    ),
+    (EntityDisabledError, "com.microsoft:entity-disabled"),
+    (PartitionNotOwnedError, "com.microsoft:partition-not-owned"),
+    (StoreLockLostError, "com.microsoft:store-lock-lost"),
+    (PublisherRevokedError, "com.microsoft:publisher-revoked"),
+    (TimeoutError, "com.microsoft:timeout"),
+    (TrackingIdProperty, "com.microsoft:tracking-id"),
+    (ProtonIo, "proton:io"),
+    (ConnectionFramingError, "amqp:connection:framing-error"),
+    (OperationCancelled, "com.microsoft:operation-cancelled"),
+    (MessageLockLost, "com.microsoft:message-lock-lost"),
+    (SessionLockLost, "com.microsoft:session-lock-lost"),
+    (
+        SessionCannotBeLocked,
+        "com.microsoft:session-cannot-be-locked"
+    ),
+    (MessageNotFound, "com.microsoft:message-not-found"),
+    (SessionNotFound, "com.microsoft:session-not-found"),
+    (EntityAlreadyExists, "com.microsoft:entity-already-exists"),
+    (ConnectionRedirect, "amqp:connection:redirect"),
+    (LinkRedirect, "amqp:link:redirect"),
+    (TransferLimitExceeded, "amqp:link:transfer-limit-exceeded"),
+    (SessionWindowViolation, "amqp:session:window-violation"),
+    (SessionErrantLink, "amqp:session:errant-link"),
+    (SessionHandleInUse, "amqp:session:handle-in-use"),
+    (SessionUnattachedHandle, "amqp:session:unattached-handle"),
+);
+
+impl From<AmqpSymbol> for AmqpErrorCondition {
+    fn from(condition: AmqpSymbol) -> Self {
+        // Note that the `from_str` implementation from `create_extensible_enum` will
+        // never return an error. So the `expect` is there to silence the compiler.
+        AmqpErrorCondition::from_str(condition.0.as_str()).expect("Invalid AMQP error condition")
```

Comment:
Then I'd call `unwrap()` to avoid the addition to the string pool.

Reply:
That makes sense. I'll fix it in the next PR.